OCP-4.X/post-install.yml

```diff
@@ -34,79 +34,110 @@
+          toggle: "{{rhcos_toggle_infra_node}}"
         - src: templates/pbench-node-machineset.yml.j2
           dest: ~/pbench-node-machineset.yml
+          toggle: "{{rhcos_toggle_pbench_node}}"
 
     - name: Get current ready node count
       shell: oc get nodes | grep " Ready" -ic
       register: rhcos_current_node_count
 
     - name: Create machinesets
       shell: |
-        oc create -f infra-node-machineset.yml,pbench-node-machineset.yml
+        oc create -f {{item.ms}}
+      when: "{{item.toggle}}"
+      with_items:
+        - ms: infra-node-machineset.yml
+          toggle: "{{rhcos_toggle_infra_node}}"
+        - ms: pbench-node-machineset.yml
+          toggle: "{{rhcos_toggle_pbench_node}}"
+
+    - name: Set expected node count
+      set_fact:
+        expected_node_count: "{{rhcos_current_node_count.stdout|int}}"
+
+    - name: Increment expected node count with infra nodes
+      set_fact:
+        expected_node_count: "{{expected_node_count|int + 3}}"
+      when: rhcos_toggle_infra_node
+
+    - name: Increment expected node count with pbench node
+      set_fact:
+        expected_node_count: "{{expected_node_count|int + 1}}"
+      when: rhcos_toggle_pbench_node
 
     - name: Poll nodes to see if creating nodes finished
       shell: oc get nodes | grep " Ready" -ic
       register: current_node_count
-      # + 3 (infra nodes) + 1 (pbench controller node)
-      until: current_node_count.stdout|int == (rhcos_current_node_count.stdout|int + 3 + 1)
+      until: current_node_count.stdout|int >= (expected_node_count|int)
       delay: 1
       retries: "{{poll_attempts|int}}"
+      when: (rhcos_toggle_infra_node == true or rhcos_toggle_pbench_node == true)
 
     - name: Relabel the infra nodes
       shell: |
         oc label nodes --overwrite -l 'node-role.kubernetes.io/infra=' node-role.kubernetes.io/worker-
+      when: rhcos_toggle_infra_node
 
     - name: Relabel the pbench controller node
       shell: |
         oc label nodes --overwrite -l 'node-role.kubernetes.io/pbench=' node-role.kubernetes.io/worker-
+      when: rhcos_toggle_pbench_node
 
     - name: Add additional label to worker nodes to provide ablity to isolate workloads on workers
       shell: |
         oc label nodes --overwrite -l 'node-role.kubernetes.io/worker=' computenode=true
 
-    - name: Disable CVO to prevent squashed configuration changes to cluster operators
-      shell: |
-        oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
+    # Due to upgrade testing we will no longer disable the CVO
+    # - name: Disable CVO to prevent squashed configuration changes to cluster operators
+    #   shell: |
+    #     oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
 
     - name: Taint the pbench controller node
       shell: |
         oc adm taint node -l node-role.kubernetes.io/pbench= dedicated=pbench:NoSchedule
```

Comment:
@akrzos can we change the taint to role=controller:NoSchedule. Also maybe we should change the name of the pbench node to controller for it project that it can host the controller pod which can run any workload using the byo mode with or without pbench. But I think we can do it later.

Reply:
Sounds good.  I'll change it.

Reply:
Ok it's changed